codex/rest/api.nim

```diff
@@ -258,6 +287,16 @@ proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
       trace "Something went wrong error"
       return RestApiResponse.error(Http500)
 
+  router.api(
+    MethodGet,
+    "/api/codex/v1/files") do () -> RestApiResponse:
+      without cids =? await node.blockStore.listBlocks(BlockType.Manifest):
+        trace "Failed to listBlocks"
+        return RestApiResponse.error(Http500)
+
+      let json = await formatManifestBlocks(node.blockStore, cids)
+      return RestApiResponse.response($json, contentType="application/json")
```

Comment:
My personal preference here would be to flatten `cids` to a `seq[Manifest]` and then all you need to do is serialize `seq[Manifest]` (having marked the needed fields with `{.serialize.}`. 
```suggestion
      without cids =? await node.blockStore.listBlocks(BlockType.Manifest):
        trace "Failed to listBlocks"
        return RestApiResponse.error(Http500)
        
      let manifests: seq[Manifest] = @[]
      for c in cids:
        if cid =? await c:
          if blk =? await store.getBlock(cid) and manifest =? Manifest.decode(blk):
            manifests.add(manifest)

      return RestApiResponse.response(manifests.toJson, contentType="application/json")
```

Of course if we don't want to mark the Manifest fields with `{.serialize.}`, I've. outlined an alternative approach in the other comment.

Reply:
Going to a seq[Manifest] and serializing it, isn't going to work: The Cid is probably the most important thing this endpoint has to return and it's (of course) not in the manifest. I hope you approve of the solution I came up with for this though. Trying to go for the best-of-both approaches.

Reply:
Exposing both the Cid and Manifest is not a problem at all, this is where creating a new type in `rest/json.nim` is appropriate:

```nim
```

Reply:
I have now applied a `RestContent` object for serialization as suggested here, to go with the /content API call.

Reply:
I did add a RestContent api type. Indeed it made things better.